openassessment/xblock/message_mixin.py

```diff
@@ -42,51 +42,55 @@ def render_message(self, data, suffix=''):
 
         is_closed = deadline_info.get('general').get('is_closed')
 
-        # Finds the status_information which describes the closed status of the current step (defaults to submission)
-        status_info = deadline_info.get(status, deadline_info.get("submission"))
-        status_is_closed = status_info.get('is_closed')
-
-        # Default context is empty
-        context = {}
-
-        # Default path leads to an "instruction-unavailable" block
-        path = 'openassessmentblock/message/oa_message_unavailable.html'
+        # Finds the deadline info of the current step (defaults to submission)
+        current_status_deadline = deadline_info.get(status, deadline_info.get("submission"))
 
         # Render the instruction message based on the status of the workflow
         # and the closed status.
         if status == "done" or status == "waiting":
             path, context = self.render_message_complete(status_details)
-        elif is_closed or status_is_closed:
-            path, context = self.render_message_closed(status_info)
-        elif status == "self":
-            path, context = self.render_message_self(deadline_info)
-        elif status == "peer":
-            path, context = self.render_message_peer(deadline_info)
-        elif status == "training":
-            path, context = self.render_message_training(deadline_info)
+        elif is_closed or current_status_deadline.get('is_closed'):
+            path, context = self.render_message_closed(current_status_deadline)
+        elif status == "self" or status == "peer" or status == "training":
+            path, context = self.render_message_incomplete(status, deadline_info)
         elif status is None:
             path, context = self.render_message_open(deadline_info)
+        else:
+            # Default path leads to an "instruction-unavailable" block
```

Comment:
There is a known issue with cancelled submissions-- https://openedx.atlassian.net/browse/TNL-3847